File: src/main/kotlin/com/github/bin/command/EqualCommand.java
package com.github.bin.command;

import com.github.bin.service.RoomConfig;
import lombok.RequiredArgsConstructor;

import java.util.Objects;

/**
 * @author bin
 * @since 2023/08/22
 */
@RequiredArgsConstructor
public abstract class EqualCommand implements Command {
    protected final String msg;

    @Override
    public final boolean invoke(RoomConfig roomConfig, String id, String msg) {
        if (Objects.equals(this.msg, msg)) {
            return invoke(roomConfig, id);
        }
        return false;
    }

    abstract boolean invoke(RoomConfig roomConfig, String id);
}

File: src/main/kotlin/com/github/bin/command/EqualCommand.kt
package com.github.bin.command

import com.github.bin.service.RoomConfig

/**
 * @author bin
 * @version 1.0.0
 * @since 2023/7/9
 */
abstract class EqualCommand(protected val msg: String) : Command {
    override fun invoke(roomConfig: RoomConfig, id: String, msg: String): Boolean {
        if (this.msg == msg) {
            return invoke(roomConfig, id)
        }
        return false
    }

    abstract fun invoke(roomConfig: RoomConfig, id: String): Boolean
}
